### PR TITLE
test: functional tests simple form 

### PR DESCRIPTION
**Pull request by gabcoelho**
### Related Issues
Closes: #1005 
Related: #500 

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 1]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
                 let url = "http://localhost:8080/simpleform"
                 self.application = TestUtils.launchBeagleApplication(url: url)
             }
         }
-        
-        Given("^the app did load simpleform screen$") { _, _ -> Void in
-            XCTAssertTrue(ScreenElements.SIMPLE_FORM_SCREEN_HEADER.element.exists)
+
+        Given("^that I'm on the simple form screen$") { _, _ -> Void in
+            XCTAssertTrue(self.application!.staticTexts[Seeds.screenTitle].exists)
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:34]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L34"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:34</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 2]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
         
-        When("^I click on text field \"([^\\\"]*)\"$") { args, _ -> Void in
-            guard let param = args?[0],
-                let text: ScreenElements = ScreenElements(rawValue: param) else {
-                    return
-            }
-            screen.clickOnText(textOption: text)
+        // Scenario 1
+        Then("^all my children components should render their respective attributes correctly$") { _, _ -> Void in
+            XCTAssertTrue(self.application!.textFields[placeholder: Seeds.emailPlaceholder]!.exists)
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:39]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L39"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:39</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 3]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            XCTAssertTrue(self.application!.textFields[placeholder: Seeds.passPlaceholder]!.exists)
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:40]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L40"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:40</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 4]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            XCTAssertTrue(self.application!.buttons[Seeds.submitButtonTitle].exists)
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:41]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L41"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:41</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 5]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
         }
-        
-        When("^insert text \"([^\\\"]*)\"$") { args, _ -> Void in
-            guard let text: String = (args?[0]) else { return }
-            screen.typeTextIntoField(insertText: text)
+
+        // Scenario 2
+        When("^I click on a textInput with email Type and type in my \"([^\\\"]*)\"$") { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:46]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L46"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:46</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 6]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+
+            let textField = self.application!.textFields[placeholder: Seeds.emailPlaceholder]!
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:48]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L48"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:48</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 7]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            textField.typeText(text)
         }
         
-        Then("^all my simple form components should render their respective text attributes correctly$") { _, _ -> Void in
-            XCTAssertTrue(ScreenElements.SIMPLE_FORM_TITLE.element.exists)
-            XCTAssertTrue(ScreenElements.ZIP_FIELD.element.exists)
-            XCTAssertTrue(ScreenElements.STREET_FIELD.element.exists)
-            
+        When("^I click on a textInput with password Type and type in my \"([^\\\"]*)\"$") { args, _ -> Void in
+            let text = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:54]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L54"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:54</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 8]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+
+            let textField = self.application!.textFields[placeholder: Seeds.passPlaceholder]!
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:56]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L56"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:56</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 9]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            textField.tap()
+            textField.typeText(text)
         }
         
-        Then("confirm popup should appear correctly$") { _, _ -> Void in
-            screen.confirmPopupCorrectly()
+        When("^I click on button with title Click to Submit") { args, _ -> Void in
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:61]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L61"></span>Unused parameter “args” in a closure should be replaced with _.<br />
<code>unused_closure_parameter</code> <code>SimpleFormSteps.swift:61</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 10]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            let button = self.application!.buttons[Seeds.submitButtonTitle]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:62]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L62"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:62</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 11]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            button.tap()
         }
+
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:66]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L66"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>SimpleFormSteps.swift:66</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 12]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
         
+        Then(#"^verify if the email: "([^\"]*)" and the password: "([^\"]*)" is appearing correctly$"#) { args, _ -> Void in
+            let email = args![0]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:68]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L68"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:68</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 13]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            let pass = args![1]
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:69]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L69"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:69</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 14]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            
+            XCTAssertTrue(self.application!.textFields[Seeds.emailInputText].exists)
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:71]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L71"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:71</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 15]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            XCTAssertTrue(self.application!.textFields[Seeds.passInputText].exists)
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:72]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L72"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:72</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 16]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+
+        }
+    }
+}
+
+// MARK: - Helpers
+fileprivate struct Seeds {
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:79]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L79"></span>Prefer <code>private</code> over <code>fileprivate</code> declarations.<br />
<code>private_over_fileprivate</code> <code>SimpleFormSteps.swift:79</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 17]
```diff
@@ -22,44 +22,73 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+    static let screenTitle = "SimpleForm"
+    static let title = "Fill the form"
+    static let emailPlaceholder = "Type in your email"
+    static let passPlaceholder = "Type in your password"
+    static let submitButtonTitle = "Click to Submit"
+    static let emailInputText = "teste@simpleform.com"
+    static let passInputText = "123"
+}
+
+
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:89]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/8307aec292afb465957a93bbffd9557ffd15e74f/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L89"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>SimpleFormSteps.swift:89</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 18]
```diff
@@ -22,44 +22,70 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            let button = self.application!.buttons[Seeds.submitButtonTitle]
+            button.tap()
         }
+
         
+        Then(#"^verify if the email: "([^\"]*)" and the password: "([^\"]*)" is appearing correctly$"#) { args, _ -> Void in
+            let email = args![0]
+            let pass = args![1]
+            
+            XCTAssertTrue(self.application!.textFields[email].exists)
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:71]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/634396e6407481fd2575bc79e6959f7e6d9321d4/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L71"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:71</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 19]
```diff
@@ -22,44 +22,70 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+            XCTAssertTrue(self.application!.textFields[pass].exists)
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:72]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/634396e6407481fd2575bc79e6959f7e6d9321d4/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L72"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:72</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 20]
```diff
@@ -22,44 +22,70 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+
+// MARK: - Helpers
+fileprivate struct Seeds {
+    static let screenTitle = "SimpleForm"
+    static let emailPlaceholder = "Type in your email"
+    static let passPlaceholder = "Type in your pass"
+    static let submitButtonTitle = "Click to Submit"
+}
+
+
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:86]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/634396e6407481fd2575bc79e6959f7e6d9321d4/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L86"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>SimpleFormSteps.swift:86</code></td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/SimpleFormScreenBuilder.kt [hunk 21]
```diff
@@ -25,8 +25,10 @@ import br.com.zup.beagle.widget.action.SetContext
 import br.com.zup.beagle.widget.action.SubmitForm
 import br.com.zup.beagle.widget.context.ContextData
 import br.com.zup.beagle.widget.context.expressionOf
+import br.com.zup.beagle.widget.context.valueOf
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/SimpleFormScreenBuilder.kt:28]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d1a41f9eb08613438b140349706687adb20a4e0/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/SimpleFormScreenBuilder.kt#L28"></span>The import ‘br.com.zup.beagle.widget.context.valueOf’ is unused.</td>
    </tr>
  </tbody>
</table>

File: backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/SimpleFormScreenBuilder.kt [hunk 22]
```diff
@@ -25,8 +25,10 @@ import br.com.zup.beagle.widget.action.SetContext
...
 import br.com.zup.beagle.widget.core.EdgeValue
 import br.com.zup.beagle.widget.core.Size
+import br.com.zup.beagle.widget.core.TextInputType
```
Comment (botbeaglezup) [inline backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/SimpleFormScreenBuilder.kt:31]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9d1a41f9eb08613438b140349706687adb20a4e0/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/SimpleFormScreenBuilder.kt#L31"></span>The import ‘br.com.zup.beagle.widget.core.TextInputType’ is unused.</td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 23]
```diff
@@ -22,44 +22,69 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
-        When("^I click on text field \"([^\\\"]*)\"$") { args, _ -> Void in
-            guard let param = args?[0],
-                let text: ScreenElements = ScreenElements(rawValue: param) else {
-                    return
-            }
-            screen.clickOnText(textOption: text)
+        // Scenario 1
+        Then("^all my children components should render their respective attributes correctly$") { _, _ -> Void in
+            XCTAssertTrue(self.application!.textFields[placeholder: Seeds.emailPlaceholder]!.exists)
+            XCTAssertTrue(self.application!.textFields[placeholder: "Type in your password"]!.exists)
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:40]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/3c526e9018d803040a204ee6cb9a1c80f887cce7/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L40"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:40</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 24]
```diff
@@ -22,44 +22,69 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
         
-        Then("^all my simple form components should render their respective text attributes correctly$") { _, _ -> Void in
-            XCTAssertTrue(ScreenElements.SIMPLE_FORM_TITLE.element.exists)
-            XCTAssertTrue(ScreenElements.ZIP_FIELD.element.exists)
-            XCTAssertTrue(ScreenElements.STREET_FIELD.element.exists)
-            
+        When("^I click on a textInput with password Type and type in my \"([^\\\"]*)\"$") { args, _ -> Void in
+            let text = args![0]
+
+            let textField = self.application!.textFields[placeholder: "Type in your password"]!
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:56]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/3c526e9018d803040a204ee6cb9a1c80f887cce7/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L56"></span>Force unwrapping should be avoided.<br />
<code>force_unwrapping</code> <code>SimpleFormSteps.swift:56</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift [hunk 25]
```diff
@@ -22,44 +22,69 @@ class SimpleFormSteps: CucumberStepsDefinition {
...
+}
+
+// MARK: - Helpers
+fileprivate struct Seeds {
+    static let screenTitle = "SimpleForm"
+    static let emailPlaceholder = "Type in your email"
+    static let submitButtonTitle = "Click to Submit"
+}
+
+
```
Comment (botbeaglezup) [inline iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift:85]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/3c526e9018d803040a204ee6cb9a1c80f887cce7/iOS/Tests/AutomatedTestsUITests/Steps/SimpleFormSteps.swift#L85"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>SimpleFormSteps.swift:85</code></td>
    </tr>
  </tbody>
</table>

**Review by igorgiumellizup — APPROVED** [review]

**Review by luisgustavozup — APPROVED** [review]

**Review by dantes-git — APPROVED** [review]

**Review by carlossteinzup — APPROVED** [review]

**Review by LG95 — APPROVED** [review]

**Pull request merged**